src/vma/dev/ring_slave.h

```diff
@@ -162,59 +162,55 @@ class ring_slave : public ring
 	ring_slave(int if_index, ring* parent, ring_type_t type);
 	virtual ~ring_slave();
 
-	virtual void print_val();
-
-	virtual void        restart();
-
 	virtual int         get_num_resources() const { return 1; };
+	virtual bool        is_up() = 0;
 	virtual bool        is_member(ring_slave* rng);
 	virtual bool        is_active_member(ring_slave* rng, ring_user_id_t id);
-	virtual ring_user_id_t	generate_id();
-	virtual ring_user_id_t	generate_id(const address_t src_mac, const address_t dst_mac, uint16_t eth_proto, uint16_t encap_proto, uint32_t src_ip, uint32_t dst_ip, uint16_t src_port, uint16_t dst_port);
-	virtual bool        is_up() = 0;
-	virtual void        inc_tx_retransmissions(ring_user_id_t id);
-	virtual void        inc_cq_moderation(size_t sz_data) = 0;
-	virtual bool        rx_process_buffer(mem_buf_desc_t* p_rx_wc_buf_desc, void* pv_fd_ready_array);
-	virtual uint32_t    get_underly_qpn() = 0;
 	virtual bool        attach_flow(flow_tuple& flow_spec_5t, pkt_rcvr_sink* sink);
 	virtual bool        detach_flow(flow_tuple& flow_spec_5t, pkt_rcvr_sink* sink);
+	virtual bool        rx_process_buffer(mem_buf_desc_t* p_rx_wc_buf_desc, void* pv_fd_ready_array);
+	virtual void        print_val();
```

Comment:
I prefer to see print_val() after constructor and destructor.

Reply:
done